# Allows FluidSynth to build with position-independent code enabled when building FluidSynth as a static library.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -135,9 +135,9 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 # Compile with position independent code if the user requested a shared lib, i.e. no PIC if static requested.
 # This is cmakes default behavior, but here it's explicitly required due to the use of libfluidsynth-OBJ as object library,
 # which would otherwise always be compiled without PIC.
-if ( NOT CMAKE_POSITION_INDEPENDENT_CODE )
-    set ( CMAKE_POSITION_INDEPENDENT_CODE ${BUILD_SHARED_LIBS} )
-endif ( NOT CMAKE_POSITION_INDEPENDENT_CODE )
+
+
+
 
 # the default global visibility level for all target
 # no visibility support on OS2
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -251,7 +251,7 @@ if ( WIN32 )
     configure_file(${FluidSynth_SOURCE_DIR}/cmake_admin/VersionResource.rc.in ${VersionFilesOutputVariable} @ONLY)
 endif ( WIN32 )
 
-add_library ( libfluidsynth-OBJ OBJECT
+add_library ( libfluidsynth STATIC
     ${config_SOURCES}
     ${fluid_alsa_SOURCES}
     ${fluid_aufile_SOURCES}
@@ -282,8 +282,8 @@ add_library ( libfluidsynth-OBJ OBJECT
     ${public_main_HEADER}
     ${VersionFilesOutputVariable}
 )
-
-target_include_directories ( libfluidsynth-OBJ PRIVATE
+add_library ( libfluidsynth-OBJ ALIAS libfluidsynth )
+target_include_directories ( libfluidsynth PRIVATE
     ${FluidSynth_BINARY_DIR}
     ${FluidSynth_BINARY_DIR}/include
     ${FluidSynth_SOURCE_DIR}/src
@@ -299,7 +299,7 @@ target_include_directories ( libfluidsynth-OBJ PRIVATE
 )
 
 if ( LIBFLUID_CPPFLAGS )
-  set_target_properties ( libfluidsynth-OBJ
+  set_target_properties ( libfluidsynth
     PROPERTIES COMPILE_FLAGS ${LIBFLUID_CPPFLAGS} )
 endif ( LIBFLUID_CPPFLAGS )
 
@@ -315,12 +315,12 @@ endif ( MACOSX_FRAMEWORK )
 # static library instead, set the option BUILD_SHARED_LIBS to FALSE.
 # Further note: The headers must be explicitly added here to have CMake install
 # them correctly in case of MACOSX_FRAMEWORK
-add_library ( libfluidsynth
-    $<TARGET_OBJECTS:libfluidsynth-OBJ>
-    ${public_main_HEADER}
-    ${public_HEADERS}
-    ${MACOSX_FRAMEWORK_SOURCE_HACK}
-)
+
+
+
+
+
+
 
 if ( MACOSX_FRAMEWORK )
     set_source_files_properties ( ${public_HEADERS}
@@ -379,7 +379,7 @@ endif ( MACOSX_FRAMEWORK )
 # See https://cmake.org/cmake/help/latest/command/target_link_libraries.html#linking-object-libraries
 # Object Libraries may "link" to other libraries to get usage requirements,
 # but since they do not have a link step nothing is done with their object files.
-target_link_libraries ( libfluidsynth-OBJ PUBLIC
+target_link_libraries ( libfluidsynth PUBLIC
     ${DART_LIBS}
     ${KAI_LIBS}
     ${COREAUDIO_LIBS}
@@ -464,12 +464,12 @@ endif ( OPENSLES_SUPPORT )
 
 # This doesn't install any object file at all!
 # it is only a trick for exporting targets
-install( TARGETS libfluidsynth-OBJ
-    EXPORT FluidSynthTargets
-    OBJECTS )
+
+
+
 
 # Here are applied/linked the OBJECT library dependencies
-target_link_libraries ( libfluidsynth PRIVATE libfluidsynth-OBJ )
+
 
 # ************ CLI program ************
 
